### PR TITLE
*: Improve clarification on transaction retry

### PR DESCRIPTION
**Pull request by huangxiuyan**
via: https://github.com/pingcap/docs/pull/1027 @CaitinChen PTAL

### CODE REVIEW

File: sql/transaction-isolation.md [hunk 1]
```diff
@@ -50,9 +52,9 @@ MySQL 的可重复读隔离级别并非 snapshot 隔离级别，MySQL 可重复
 
 ## 事务重试
 
-对于 insert/delete/update 操作，如果事务执行失败，并且系统判断该错误为可重试，会在系统内部自动重试事务。
+执行失败的事务可能会由 TiDB 自动重试，这可能会导致更新丢失。通过设置 `tidb_disable_txn_auto_retry=TRUE` 和 `tidb_retry_limit = 0` 可关掉该项功能。
```
Comment (jackysp) [inline sql/transaction-isolation.md]:
```suggestion
执行失败的事务可能会由 TiDB 自动重试，这可能会导致更新丢失。通过设置 `tidb_disable_txn_auto_retry=on` 和 `tidb_retry_limit = 0` 可关掉该项功能。
```

File: sql/transaction-isolation.md [hunk 2]
```diff
@@ -50,9 +52,9 @@ MySQL 的可重复读隔离级别并非 snapshot 隔离级别，MySQL 可重复
...
 
-通过配置参数 `retry-limit` 可控制自动重试的次数：
+另外，通过配置参数 `retry-limit` 可控制自动重试的次数：
```
Comment (jackysp) [inline sql/transaction-isolation.md]:
```suggestion
```

File: sql/transaction-isolation.md [hunk 3]
```diff
@@ -89,15 +91,16 @@ retry-limit = 10
 | `}` | |
 | `commit;` // 自动重试 | |
 
-因为 TiDB 自动重试机制会把事务第一次执行的所有语句重新执行一遍，当一个事务里的后续语句是否执行取决于前面语句执行结果的时候，自动重试无法保证最终结果符合预期。这种情况下，需要在应用层重试整个事务。
+因为 TiDB 自动重试机制会把事务第一次执行的所有语句重新执行一遍，当一个事务里的后续语句是否执行取决于前面语句执行结果的时候，自动重试会违反快照隔离，导致更新丢失。这种情况下，需要在应用层重试整个事务。
 
-通过配置全局变量 `tidb_disable_txn_auto_retry` 可以关掉显式事务的重试。
+通过配置 `tidb_disable_txn_auto_retry` 和 `tidb_retry_limit` 变量可以关掉显式事务的重试。
```
Comment (jackysp) [inline sql/transaction-isolation.md]:
```suggestion
通过配置 `tidb_disable_txn_auto_retry` 和 `tidb_retry_limit` 变量可以关掉事务的重试。
```

File: sql/transaction-isolation.md [hunk 4]
```diff
@@ -89,15 +91,16 @@ retry-limit = 10
...
 
-```
-set @@global.tidb_disable_txn_auto_retry = 1;
+```sql
+SET GLOBAL tidb_disable_txn_auto_retry = TRUE;
```
Comment (jackysp) [inline sql/transaction-isolation.md]:
```suggestion
```

File: sql/transaction-isolation.md [hunk 5]
```diff
@@ -89,15 +91,16 @@ retry-limit = 10
...
+SET GLOBAL tidb_retry_limit = 0;
 ```
 
-这个变量不会影响 `auto_commit = 1` 的单语句的隐式事务，仍然会自动重试。
+改变 `tidb_disable_txn_auto_retry` 变量不会影响     `auto_commit = 1` 的单语句的隐式事务，因为该语句仍然会自动重试。
```
Comment (jackysp) [inline sql/transaction-isolation.md]:
```suggestion
改变 `tidb_disable_txn_auto_retry` 变量不会影响 `auto_commit = 1` 的单语句的隐式事务，因为该语句仍然会自动重试。
```

File: sql/transaction-model.md [hunk 6]
```diff
@@ -9,6 +9,10 @@ TiDB 使用乐观事务模型，在执行 `Update`、`Insert`、`Delete` 等语
 
 ## 与 MySQL 行为及性能对比
 
+### 事务重试
+
+执行失败的事务默认由 TiDB 自动重试，这会导致更新丢失。可通过配置 `tidb_disable_txn_auto_retry = TRUE` 和 `tidb_retry_limit = 0` 关掉该项功能。
```
Comment (jackysp) [inline sql/transaction-model.md]:
```suggestion
执行失败的事务默认由 TiDB 自动重试，这会导致更新丢失。可通过配置 `tidb_retry_limit = 0` 关掉该项功能。
```

File: sql/transaction-isolation.md [hunk 7]
```diff
@@ -61,9 +63,9 @@ MySQL 的可重复读隔离级别并非 snapshot 隔离级别，MySQL 可重复
 retry-limit = 10
 ```
 
-## 乐观事务注意事项
+## 自动重试导致的事务异常
 
-因为 TiDB 使用乐观锁机制，通过显式的 `BEGIN` 语句创建的事务，在遇到冲突后自动重试可能会导致最终结果不符合预期。
+因为 TiDB 会[默认](#事务重试)自动重试事务，通过显式的 `BEGIN` 语句创建的事务，在遇到冲突后自动重试可能会导致最终结果不符合预期。
```
Comment (yikeke) [inline sql/transaction-isolation.md]:
```suggestion
因为 TiDB 会[默认](#事务重试)自动重试事务，所以如果由显式的 `BEGIN` 语句创建的事务在遇到冲突后自动重试，可能会导致最终结果不符合预期。
```

**Comment by CaitinChen** [comment]
@jackysp @yikeke PTAL again. Thanks.

**Comment by CaitinChen** [comment]
@jackysp PTAL again. Thanks.

**Comment by jackysp** [comment]
I think we should address the comments first. @CaitinChen

**Comment by CaitinChen** [comment]
@jackysp Comments have been addressed by @huangxiuyan

**Comment by huangxiuyan** [comment]
Sorry, my fault, i'll address the comment immediately.

**Comment by huangxiuyan** [comment]
@CaitinChen @jackysp PTAL, thanks!

File: sql/transaction-isolation.md [hunk 8]
```diff
@@ -89,15 +90,15 @@ retry-limit = 10
...
-因为 TiDB 自动重试机制会把事务第一次执行的所有语句重新执行一遍，当一个事务里的后续语句是否执行取决于前面语句执行结果的时候，自动重试无法保证最终结果符合预期。这种情况下，需要在应用层重试整个事务。
+因为 TiDB 自动重试机制会把事务第一次执行的所有语句重新执行一遍，当一个事务里的后续语句是否执行取决于前面语句执行结果的时候，自动重试会违反快照隔离，导致更新丢失。这种情况下，需要在应用层重试整个事务。
 
-通过配置全局变量 `tidb_disable_txn_auto_retry` 可以关掉显式事务的重试。
+通过配置 `tidb_retry_limit` 变量可以关掉事务的重试。
 
-```
-set @@global.tidb_disable_txn_auto_retry = 1;
+```sql
+SET GLOBAL tidb_disable_txn_auto_retry = TRUE;
```
Comment (jackysp) [inline sql/transaction-isolation.md]: SET GLOBAL tidb_retry_limit = 0;

Comment (jackysp) [inline sql/transaction-isolation.md]:
```suggestion
SET GLOBAL tidb_retry_limit = 0;
```

**Comment by huangxiuyan** [comment]
@jackysp @CaitinChen Already fixed, PTAL

**Pull request merged**